add multi-tenancy public tutorial (#2393)
giantswarm/docs: src/content/overview/architecture/authentication/index.md

We utilize Kubernetes-native RBAC to control access to resources in the platform API. Resources are segregated into organizations, each represented by a dedicated namespace, enabling improved access control. This approach allows for fine-grained permissions at both the organization and namespace levels. For more detailed information on this topic, you can refer to our comprehensive [multi-tenancy documentation]({{< relref "/overview/fleet-management/multi-tenancy" >}}).

> ### Observability Platform API

Author: @stone-z I think this should address your comment as this is where grafana organizations are introduced in our public docs.

I am not really sure about the overall structure withthe whole platform API above. @pipo02mix maybe you have some thoughts on this?

Reviewer: I am not sure if this paragraph will fit better here -> https://docs.giantswarm.io/overview/architecture/#observability

We don't explain anything special about authN for the o11y part

Author: The goal is more (as asked by Zach) to let the platform team knows what kind of RBAC they have to create when they are thinking about cluster organizations but maybe there is a better place for this?

Author: It feels a bit weird to me that we call this the platform api which is not working with the concept of multiple platforms we are building when the existing content is about cluster management.

Maybe we should rewrite the whole page to talk about the cluster management platform api and the observability platform api?

Author: Sure i understand this :) 
What I mean is that we are building a platform (Giant Swarm) to build platforms on top of it.

As of today, we have the Kaas Platform and the observability and the security platform built on top of it that makes up the Giant Swarm platform.

Right now the API is only kubernertes CRDs as the API,  that is true, but this might change with things like backstage or clickops in Grafana for instance.

Now what I'm really confused about is that the multi-tenancy page you linked is only about fleet management tenancy (reasons as to why it is under the fleet-management section) while grafana organizations is a different mechanism that is in no way related to fleets or clusters but is more linked towards projects or teams and there is currently no place for this.

I do agree with Zach that we need to have a central place to introduce the multi-tenancy mechanisms we have in-place in our platform and I thought this page could be it but maybe we shoulld not talk about organizations here at all and introduce a new page that explains the multi-tenancy concepts like cluster organizations and grafana organizations?

Reviewer: ok I got you, yeah I think we need an overview page where we talk about the different options platform engineers have with GS to manage users/tenants. I create a ticket to add that info

Author: Yes something like a multi-tenancy page would be cool.

In that case, would you prefer that I remove the whole part about the observability platform api from here and we work on a better one in the sig-docs ticket?

> The platform API provides read access to the observability tools, like the Grafana instance running the management cluster, so that you can browse across the data collected in your managed clusters and any additional data you choose to push to the platform.  For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).

Author: ```suggestion
The platform API provides read access to the observability tools, like the Grafana instance running the management cluster, so that you can browse across the data collected in your managed clusters and any additional data you choose to push to the platform. For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).
```

> The observability platform API operating on the management cluster provides you with read access (through the management cluster Grafana) to the observability data collected in managed clusters as well as any additional observability data you choose to ingest in the observability platform. For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).

Reviewer 2: Hmm it's not really an API yet with which customers can interact, right? I mean yeah we're working on that but at this point I think this part is a bit confusing. I would push more for something like "you can access our grafana with our sso" - but I'm not sure this adds any value here, wdyt?

Author: The GrafanaOrganizations CRs are an API though right?

Reviewer 2: Well...technically yes... :D If you think this is necessary or adds value then let's keep it in.

> The observability platform API operating on the management cluster provides you with read access (through the management cluster Grafana) to the observability data collected in managed clusters as well as any additional observability data you choose to ingest in the observability platform. For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).

Reviewer: ```suggestion
The platform API provides read access to the observability tools, like the Grafana instance running the management cluster, so that you can browse across the data collected in your managed clusters and any additional data you choose to push to the platform.  For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).
```
my recommendation, feel free to modify

> ### Observability Platform API
> 
> The observability platform API operating on the management cluster provides you with read access (through the management cluster Grafana) to the observability data collected in managed clusters as well as any additional observability data you choose to ingest in the observability platform. For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).

Reviewer: ```suggestion
### Observability

The observability platform API operating on the management cluster provides you with read access (through the management cluster Grafana) to the observability data collected in managed clusters as well as any additional observability data you choose to ingest in the observability platform. For more detailed information on this topic, you can refer to our comprehensive [grafana organization documentation]({{< relref "/tutorials/observability/multi-tenancy/creating-grafana-organization" >}}).
```
